### DISCUSSION TITLE
CreateBasicPublishBatch() has been removed from IChannel version 7.x

### DISCUSSION THREAD

**Discussion by cleberMargarida** [rabbitmq/rabbitmq-dotnet-client#1509] [Q&A] [answered]
`IModel` has been renamed to `IChannel` and the public API `CreateBasicPublishBatch()` has been removed from version 7.x

I would like to know, if there is any way to use `BasicPublishAsync` batched allowing messages to be grouped together in a single write(2) to the socket, reducing the I/O usage.

Why is it removed? Any concerns when using batched publishing?

Thanks in advance.

**Comment by lukebakken** [comment, marked as answer]
> BasicPublishAsync batched allowing messages to be grouped together in a single write(2) to the socket, reducing the I/O usage.

No, and that's not how `CreateBasicPublishBatch` works in 6.x, anyway.

> Why is it removed? Any concerns when using batched publishing?

It was removed because it was a convenience method that basically worked like `BasicPublish` in a loop.

One of the goals for version 7 is to reduce the amount of code in this library that must be supported.
